### PR TITLE
ci: Enable more packages in the DVSDK CI pipeline

### PR DESCRIPTION
**Pull request by chuckatkins**

### CODE REVIEW

**Comment by chuckatkins** [comment]
@cyrush @mclarsen the `dray` version constraints should get ascent building again and enabling `+ascent` in the `dvsdk`  pipeline should ensure it stays building on develop.  Please confirm these are correct.

**Comment by cyrush** [comment]
@chuckatkins thanks for working on this.

Is it possible to apply the raja and umpire constraints in the CI `spack.yaml`, instead of the `package.py`?

I think those are what we need for Ascent 0.7.2, however they will undermine building `ascent@develop`.

Or maybe we could add those for ascent less or equal to 0.7.2?

**Comment by cyrush** [comment]
(that assumes spack is treating `develop` as newer than `0.7.2`) 

I think they do, however I don't recall the details from when the added logic for this.

**Comment by chuckatkins** [comment]
> Is it possible to apply the raja and umpire constraints in the CI `spack.yaml`, instead of the `package.py`?

That would only have it pass ci but not actually work when an end user tries to install it.  If the constraint is real then it should be encoded in the `package.y`


> I think those are what we need for Ascent 0.7.2, however they will undermine building `ascent@develop`.
> Or maybe we could add those for ascent less or equal to 0.7.2?

That's a better approach.  I'll apply it as such.


> (that assumes spack is treating `develop` as newer than `0.7.2`)
> I think they do, however I don't recall the details from when the added logic for this.

It does.

The constraints are specifically for `dray`, not `ascent`, so I'll adjust the dependencies to apply up to the latest version and not develop.

**Comment by chuckatkins** [comment]
@scottwittenburg This turns on the floodgates for the DVSDK's CI pipeline

**Comment by alalazo** [comment]
As if I approved this PR. I'm not hitting the "Approve" button since auto-merge is on and I'd like @scottwittenburg to have a look at this too.

**Review by cyrush — APPROVED** [review]
@chuckatkins  -- Looks good, thanks for making those adjustments!

**Comment by chuckatkins** [comment]
> As if I approved this PR. I'm not hitting the "Approve" button since auto-merge is on and I'd like @scottwittenburg to have a look at this too.

@scottwittenburg and I talked about it and just wanted to wait until the llvm issue was resolved on devel so it's good to go.

File: var/spack/repos/builtin/packages/dray/package.py [hunk 1]
```diff
@@ -79,6 +80,7 @@ class Dray(Package, CudaPackage):
     depends_on("raja~cuda~openmp~shared", when="~cuda~openmp~shared")
     depends_on("raja~cuda+openmp~shared", when="~cuda+openmp~shared")
 
+    depends_on("umpire@:4.9", when="@:0.1.6")
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/dray/package.py:83]: @chuckatkins @cyrush I have built `dray@0.1.6` with `umpire@5.1.0`, is there a reason that wouldn't generally work?

Comment (cyrush) [reply var/spack/repos/builtin/packages/dray/package.py:83]: it's possible umpire 5 is fine, but umpire 6 definitely has API breaking changes (addressed in newer versions of dray)

**Pull request merged**